katsdptelstate/rdb_reader.py

```diff
@@ -50,6 +53,18 @@ def end_sorted_set(self, key):
         self._zset = {}
 
 
+def _parse_rdb_file(parser, fd, filename=None):
+    """Apply RDB parser to file descriptor, raising RdbParseError on error."""
+    try:
+        parser.parse_fd(fd)
+    except Exception as exc:
+        if exc.args == ('verify_magic_string', 'Invalid File Format'):
+            name = repr(filename) if filename else 'object'
+            err = RdbParseError('Invalid RDB file {}'.format(name))
+            raise_from(err, exc)
+        raise
```

Comment:
I'm going to exclude callback client errors by setting a flag when inside the callback and not recasting exceptions if they originate from the callback. Otherwise, I'll catch all exceptions in the parser, but use `raise_from` to enhance debugging info. I expect a mix of customised base `Exception` and `struct.error` exceptions if the parsing fails.